test/test_support.ts

```diff
@@ -279,56 +243,63 @@ function extractInlineSourceMap(source: string): string {
 }
 
-function compileWithTsickleCompilerHost(
-    resolvedSources: Map<string, string>, fileNames: string[], tsickleHost: tsickle.TsickleHost,
-    compilerOptions: ts.CompilerOptions):
-    {jsFiles: Map<string, string>, diagnostics: ts.Diagnostic[]} {
-  const closureJSOPtions = {
-    files: resolvedSources,
-    tsickleHost,
-    tsicklePasses: [tsickle.Pass.DECORATOR_DOWNLEVEL, tsickle.Pass.CLOSURIZE],
-  };
+export function getSourceMapWithName(
+    filename: string, files: Map<string, string>): BasicSourceMapConsumer {
+  return sourceMapTextToConsumer(getFileWithName(filename, files));
+}
 
+export function compileWithTsickleCompilerHost(
+    sources: Map<string, string>, compilerOptions: ts.CompilerOptions,
+    tsickleHost: tsickle.TsickleHost = createTsickleHost(sources)) {
+  const fileNames = toArray(sources.keys());
   const diagnostics: ts.Diagnostic[] = [];
-  const closure =
-      toClosureJS(compilerOptions, fileNames, {isTyped: true}, diagnostics, closureJSOPtions);
-  return {jsFiles: closure ? closure.jsFiles : new Map<string, string>(), diagnostics};
+  const closure = toClosureJS(
+      compilerOptions, fileNames, {isTyped: true}, diagnostics,
+      createHost(sources, compilerOptions), tsickleHost,
+      [tsickle.Pass.DECORATOR_DOWNLEVEL, tsickle.Pass.CLOSURIZE]);
+
+  expect(diagnostics).lengthOf(0, tsickle.formatDiagnostics(diagnostics));
+  return {files: closure!.jsFiles, externs: closure!.externs};
 }
 
-function compileWithTransfromer(
-    resolvedSources: Map<string, string>, fileNames: string[],
-    transformerHost: tsickle.TransformerHost, compilerOptions: ts.CompilerOptions) {
-  const jsFiles = new Map<string, string>();
-  const tsHost = createSourceReplacingCompilerHost(
-      resolvedSources,
-      createOutputRetainingCompilerHost(jsFiles, ts.createCompilerHost(compilerOptions)));
+export function compileWithTransfromer(
+    sources: Map<string, string>, compilerOptions: ts.CompilerOptions,
+    transformerHost: tsickle.TransformerHost = createTsickleHost(sources)) {
+  const fileNames = toArray(sources.keys());
+  const files = new Map<string, string>();
+  const tsHost = createOutputRetainingCompilerHost(files, createHost(sources, compilerOptions));
   const program = ts.createProgram(fileNames, compilerOptions, tsHost);
-
-  const allDiagnostics: ts.Diagnostic[] = [];
-  allDiagnostics.push(...ts.getPreEmitDiagnostics(program));
-  if (allDiagnostics.length === 0) {
-    const {diagnostics} = tsickle.emitWithTsickle(
-        program, transformerHost, {
-          transformDecorators: true,
-          transformTypesToClosure: true,
-          googmodule: true,
-          es5Mode: false,
-          untyped: false,
-        },
-        tsHost, compilerOptions);
-    allDiagnostics.push(...diagnostics);
-  }
-  return {jsFiles, diagnostics: allDiagnostics};
+  expect(ts.getPreEmitDiagnostics(program))
+      .lengthOf(0, tsickle.formatDiagnostics(ts.getPreEmitDiagnostics(program)));
```

Comment:
`.to.be.empty` is more idiomatic I think, or `.to.equal`. You should just avoid `.not.to.be.empty`, as the failure message is much better with a positive assertion.